tuplex/core/src/physical/ResultSet.cc

```diff
@@ -138,8 +138,7 @@ namespace tuplex {
                 auto num_rows = first->getNumRows();
                 // how many left to retrieve?
                 auto num_to_retrieve_from_partition = std::min(limit - i, num_rows - _curRowCounter);
```

Comment:
sure there's no underflow/overflow occuring here? subtracting two size_t can lead to bad results. maybe assert i <= limit etc or explicitly check with an if statement and throw exception

Reply:
Agree, I changed i and j to size_t from int also